api/utilities.py

```diff
@@ -52,21 +52,18 @@ def get_datetime_range(datetime_string: str | None) -> Tuple[Timestamp, Timestam
     return start_datetime, end_datetime
 
 
+@cached(ttl=600)
 async def get_current_parameter_names(ttl_hash=None):
     """
     This function get a set of standard_names currently in the datastore
     The ttl_hash should be a value that is updated at the same frequency
     we want the lru_cache to be valid for.
     """
 
-    @lru_cache(maxsize=1)
-    async def async_helper(ttl_hash):  # pylint: disable=unused-arguement
-        unique_parameter_names = dstore.GetTSAGRequest(attrs=["parameter_name"])
-        unique_parameter_names = await getTSAGRequest(unique_parameter_names)
+    unique_parameter_names = dstore.GetTSAGRequest(attrs=["parameter_name"])
+    unique_parameter_names = await getTSAGRequest(unique_parameter_names)
 
-        return set([i.combo.standard_name for i in unique_parameter_names.groups])
-
-    return await async_helper(ttl_hash)
+    return set([i.combo.parameter_name for i in unique_parameter_names.groups])
 
 
 async def verify_parameter_names(parameter_names: list) -> None:
```

Comment:
Do we want to use this, what is the advantage over just passing it into the store?

And if we do, do we want to add caching? How long does the GetTSAGRequest take?
I always try to avoid using caching. In this cache, the cached verify_parameter_names might actually be out of alignment with the parameters available in the call into the store (as that is not cached).

Reply:
If gRPC would report that the parameter name given had no matches, this becomes unnecessary.
I would like to differentiate for the user between parameter names with errors, and if the chosen query just had no matches.

Reply:
Ok, fair enough, we can keep it. What about the second paragraph of my comment? Is the potential inconsistency worth the performance improvement?

Call to get the valid parameters takes 2 ms.

Reply:
I will update the function and remove the cache.